### PR TITLE
Fix link rewrites

### PR DESCRIPTION
**Pull request by AlanGreene**
# Changes

Resolves https://github.com/tektoncd/website/issues/500
Resolves https://github.com/tektoncd/website/issues/499

Update the sync script to ensure both links and images are processed for URL rewrites.

/kind bug

# Submitter Checklist

These are the criteria that every PR should meet, please check them off as you
review them:

- [ ] Includes [docs](https://github.com/tektoncd/community/blob/master/standards.md#principles) (if user facing)
- [x] Commit messages follow [commit message best practices](https://github.com/tektoncd/community/blob/master/standards.md#commit-messages)

_See [the contribution guide](https://github.com/tektoncd/website/blob/master/CONTRIBUTING.md)
for more details._

### CODE REVIEW

File: sync/sync.py [hunk 1]
```diff
@@ -239,11 +239,13 @@ def transform_links_doc(text, base_path, local_files, rewrite_path, rewrite_url)
...
     links = get_links(text)
     # Rewrite map, only use links with href and src
-    rewrite_map = {x.get("href"): transform_link(x.get("href"), base_path, local_files, rewrite_path, rewrite_url)
+    rewrite_map_links = {x.get("href"): transform_link(x.get("href"), base_path, local_files, rewrite_path, rewrite_url)
         for x in links if x.get("href")}
-    rewrite_map = {x.get("src"): transform_link(x.get("src"), base_path, local_files, rewrite_path, rewrite_url)
+    rewrite_map_images = {x.get("src"): transform_link(x.get("src"), base_path, local_files, rewrite_path, rewrite_url)
         for x in links if x.get("src")}
 
+    rewrite_map = rewrite_map_links | rewrite_map_images
```
[the remark this answers is not in the document]
Comment (AlanGreene) [reply sync/sync.py]: Even if they do have overlapping keys they would be transformed in the same way so it shouldn't make a difference.

[the remark this answers is not in the document]
Comment (AlanGreene) [reply sync/sync.py]: Also switched to older syntax since I couldn't get netlify to use python 3.9 (by updating runtime.txt) for the `|` merge operator.

**Review by geriom — COMMENTED** [review]
/lgtm

**Review by afrittoli — COMMENTED** [review]
/approve

**Comment by AlanGreene** [comment]
Looks like this also resolved https://github.com/tektoncd/website/issues/498 🎉

**Pull request merged**